src/PerformanceMonitor/BlockHelper.cs

```diff
@@ -0,0 +1,114 @@
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using System.Collections.Generic;
+
+namespace Neo.Plugins
+{
+    internal static class BlockHelper
+    {
+        /// <summary>
+        /// Returns the time the given block was active
+        /// </summary>
+        /// <param name="block">
+        /// The given block to verify the time
+        /// </param>
+        /// <returns>
+        /// Returns 0 if <paramref name="block"/> is the current block or it is the genesis block or it is null;
+        /// otherwise, returns the time the block was active in milliseconds
+        /// </returns>
+        public static ulong GetTime(this Block block)
+        {
+            ulong time = 0;
+
+            if (block != null && block.Index > 0 && block.Index < Blockchain.Singleton.Height)
+            {
+                var nextBlock = Blockchain.Singleton.GetBlock(block.Index + 1);
+
+                if (nextBlock != null)
+                {
+                    time = nextBlock.Timestamp - block.Timestamp;
+                }
+            }
+
+            return time;
+        }
+
+        /// <summary>
+        /// Returns the average time the latest blocks are active
+        /// </summary>
+        /// <param name="desiredCount">
+        /// The desired number of blocks that should be checked to calculate the average time
+        /// </param>
+        /// <returns>
+        /// Returns the average time per block in milliseconds if the number of analysed blocks
+        /// is greater than zero; otherwise, returns 0.0
+        /// </returns>
+        public static double GetAverageTimePerBlock(this SnapshotView snapshot, uint desiredCount)
```

Comment:
As I mentioned above, I see a logic like this more useful:

```cs
private JObject GetBlocksTime(uint nBlocks, uint lastHeight)
        {
            // It is currently limited to query blocks generated in the last 24hours (86400 seconds)
            uint maxNBlocksPerDay = 86400 / (Blockchain.MillisecondsPerBlock / 1000);
            if (lastHeight != 0)
            {
                if (lastHeight >= Blockchain.Singleton.Height)
                {
                    JObject json = new JObject();
                    return json["error"] = "Requested height to start " + lastHeight + " exceeds " + Blockchain.Singleton.Height;
                }

                if (nBlocks > lastHeight)
                {
                    JObject json = new JObject();
                    return json["error"] = "Requested " + nBlocks + " blocks timestamps is greater than starting at height " + lastHeight;
                }
            }

            if (nBlocks > maxNBlocksPerDay)
            {
                JObject json = new JObject();
                return json["error"] = "Requested number of blocks timestamps exceeds " + maxNBlocksPerDay;
            }

            if (nBlocks >= Blockchain.Singleton.Height)
            {
                JObject json = new JObject();
                return json["error"] = "Requested number of blocks timestamps " + nBlocks + " exceeds quantity of known blocks " + Blockchain.Singleton.Height;
            }

            if (nBlocks == 0)
            {
                JObject json = new JObject();
                return json["error"] = "Requested number of block times can not be = 0";
            }

            JArray array = new JArray();
            uint heightToBegin = lastHeight > 0 ? lastHeight - nBlocks : (Blockchain.Singleton.Height - 1) - nBlocks;
            for (uint i = heightToBegin; i <= heightToBegin + nBlocks; i++)
            {
                Header header = Blockchain.Singleton.GetHeader(i);
                if (header == null) break;

                JObject json = new JObject();
                json["timestamp"] = header.Timestamp;
                json["height"] = i;
                array.Add(json);
            }

            return array;
        }
```

You can pass begin and count.
This code can be adjusted for average if you do not want to return an array of values, which I see as useful.